### ISSUE TITLE
`prefer-https` false positive on XML namespace URIs (e.g. SVG xmlns)

### ISSUE THREAD

**Issue by yoyo837** [sindresorhus/eslint-plugin-unicorn#3062]
## Description

The `prefer-https` rule reports `http://` URLs in XML namespace URIs like SVG's `xmlns="http://www.w3.org/2000/svg"`. These are **namespace identifiers**, not network request addresses — they should not be converted to HTTPS.

## Example

```jsx
// JSX / TSX
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="..." />
</svg>
```

This triggers `unicorn/prefer-https` on `http://www.w3.org/2000/svg`, with an autofix that changes it to `https://www.w3.org/2000/svg`.

## Why this is a false positive

XML namespace URIs are **opaque identifiers** defined by the W3C spec. The namespace `http://www.w3.org/2000/svg` is the canonical, standards-defined string that XML/HTML parsers match against. While `https://www.w3.org/2000/svg` happens to redirect in browsers, the namespace spec requires the exact `http://` URI — some strict XML processors may not recognize the `https://` variant.

Similar standard namespace URIs that would be affected:
- `http://www.w3.org/1999/xlink`
- `http://www.w3.org/1999/xhtml`
- `http://www.w3.org/XML/1998/namespace`
- `http://schemas.xmlsoap.org/soap/envelope/`

## Suggested fix

The rule should skip `http://` URLs that appear inside `xmlns` attribute values (in JSX/HTML-like contexts), or maintain a list of well-known namespace URI patterns to ignore.

## Environment

- eslint-plugin-unicorn version: latest

**Comment by yoyo837** [comment]
I've prepared a fix in my fork: [fix/prefer-https-xmlns](https://github.com/yoyo837/eslint-plugin-unicorn/tree/fix/prefer-https-xmlns)

The fix is minimal — a single function that checks if the `http://` URL is preceded by a `xmlns` or `xmlns:prefix` attribute:

```js
function isXmlNamespaceUri(text, matchIndex) {
    const preceding = text.slice(Math.max(0, matchIndex - 20), matchIndex);
    return /xmlns(?::\w+)?\s*=\s*["']?$/i.test(preceding);
}
```

This handles all namespace declarations regardless of tag name (svg, html, custom elements, prefixed namespaces like `xmlns:ns`).

Test cases added to `valid`:
- `` `<svg xmlns="http://www.w3.org/2000/svg"></svg>` ``
- `` `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">...</svg>` `` (JSX)
- `` `<html xmlns="http://www.w3.org/1999/xhtml"></html>` ``
- `` `<my:tag xmlns:ns="http://example.com/ns"></my:tag>` ``

All 36 tests pass (32 existing + 4 new).

I'm unable to open a PR due to the repository's collaborator-only PR restriction. Happy to submit a PR if granted access, or feel free to cherry-pick the changes from the fork.
